File: src/main/java/ar/edu/um/tesoreria/rest/model/Facultad.java
/**
 * 
 */
package ar.edu.um.tesoreria.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Table
@EqualsAndHashCode(callSuper=false)
@NoArgsConstructor
@AllArgsConstructor
public class Facultad extends Auditable implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = -1370051256170942446L;

	@Id
	@Column(name = "fac_id")
	private Integer facultadId;

	@Column(name = "fac_nombre")
	private String nombre = "";

	@Column(name = "fac_codigo_empresa")
	private String codigoempresa = "";

	@Column(name = "fac_server")
	private String server = "";

	@Column(name = "fac_db_adm")
	private String dbadm = "";

	@Column(name = "fac_dsn")
	private String dsn = "";

	@Column(name = "fac_pla_cuenta")
	private BigDecimal cuentacontable = new BigDecimal(0);

	@Column(name = "api_server")
	private String apiserver = "";

	@Column(name = "api_port")
	private Long apiport = 0L;

}

File: src/main/java/um/tesoreria/core/kotlin/model/Facultad.kt
package um.tesoreria.core.kotlin.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import java.math.BigDecimal

@Entity
data class Facultad(

    @Id
    @Column(name = "fac_id")
    var facultadId: Int? = null,

    @Column(name = "fac_nombre")
    var nombre: String? = "",

    @Column(name = "fac_codigo_empresa")
    var codigoempresa: String? = "",

    @Column(name = "fac_server")
    var server: String? = "",

    @Column(name = "fac_db_adm")
    var dbadm: String? = "",

    @Column(name = "fac_dsn")
    var dsn: String? = "",

    @Column(name = "fac_pla_cuenta")
    var cuentacontable: BigDecimal? = BigDecimal.ZERO,

    @Column(name = "api_server")
    var apiserver: String? = "",

    @Column(name = "api_port")
    var apiport: Long? = 0L

) : Auditable()
